packages/coin-sol/src/utils/transactionUtil.ts

```diff
@@ -0,0 +1,156 @@
+import * as params from '../config/params';
+import { TransactionOptions, TransactionType } from '../config/types';
+import { addressToHex, encodeLength, isBase58Format, numberToStringHex } from './stringUtil';
+
+import base58 from 'bs58';
+
+const evenHexDigit = (hex: string) => (hex.length % 2 !== 0 ? `0${hex}` : hex);
+
+export function pubKeyToAddress(publicKey: string): string {
+  const pubKeyBuf = Buffer.from(publicKey, 'hex');
+  return base58.encode(pubKeyBuf);
+}
+
+export class RawTransaction {
+  sigCount: string;
+  signature: string;
+  numberRequireSignature: string;
+  numberReadonlySignedAccount: string;
+  numberReadonlyUnSignedAccount: string;
+  keyCount: string;
+  keys: string;
+  recentBlockHash: string;
+  instructionCount: string;
+  programIdIndex: string;
+  keyIndicesCount: string;
+  keyIndices: string;
+  dataLength: string;
+  data: string;
+
+  constructor(rawTx: {
+    signature?: string;
+    numberRequireSignature: string;
+    numberReadonlySignedAccount: string;
+    numberReadonlyUnSignedAccount: string;
+    keyCount: string;
+    keys: string;
+    recentBlockHash: string;
+    instructionCount: string;
+    programIdIndex: string;
+    keyIndicesCount: string;
+    keyIndices: string;
+    dataLength: string;
+    data: string;
+  }) {
+    this.sigCount = '01';
+    this.signature = rawTx.signature || '';
+    this.numberRequireSignature = rawTx.numberRequireSignature;
+    this.numberReadonlySignedAccount = rawTx.numberReadonlySignedAccount;
+    this.numberReadonlyUnSignedAccount = rawTx.numberReadonlyUnSignedAccount;
+    this.keyCount = rawTx.keyCount;
+    this.keys = rawTx.keys;
+    this.recentBlockHash = rawTx.recentBlockHash;
+    this.instructionCount = rawTx.instructionCount;
+    this.programIdIndex = rawTx.programIdIndex;
+    this.keyIndicesCount = rawTx.keyIndicesCount;
+    this.keyIndices = rawTx.keyIndices;
+    this.dataLength = rawTx.dataLength;
+    this.data = rawTx.data;
+  }
+
+  dataEncode(amount: number | string, decimals?: number | undefined): string {
+    const programIdToNumber = Number(this.programIdIndex);
+    const isNormalTransfer = programIdToNumber === 2;
+    const dataAlloc = isNormalTransfer ? 12 : 9;
+    const data = Buffer.alloc(dataAlloc);
+
+    const v2e32 = Math.pow(2, 32);
+
+    const value = Number(amount) * (decimals ? Math.pow(10, decimals) : params.LAMPORTS_PER_SOL);
+
+    const hi32 = Math.floor(value / v2e32);
+    const lo32 = value - hi32 * v2e32;
+
+    const programIdIndexSpan = isNormalTransfer ? 4 : 1;
+
+    data.writeUIntLE(programIdToNumber, 0, programIdIndexSpan);
+    data.writeUInt32LE(lo32, programIdIndexSpan);
+    data.writeInt32LE(hi32, programIdIndexSpan + 4);
```

Comment:
Shouldn't it be `data.writeUInt32LE`? It seems that all `data` layout is `BufferLayout.u64`.  
Please point out if I missing anything.

Reply:
@hank121314  This one we just need to encode each data, the other fields don't need to be used, so we don't have to import BufferLayout

Reply:
I mean `hi32` and `lo32` will equal `amount`. 
It will always be an unsigned Integer.
Both `MintToInstructionData ` and `Transfer` are encoded amount as `BufferLayout.u64`.
And `BufferLayout.u64` is encoded in this way https://github.com/pabigot/buffer-layout/blob/326345b0c87d2a98116b22a236ea6623180ffbb1/lib/Layout.js#L750-L759.
So shouldn't it be `data.writeUInt32LE`?